### PR TITLE
Better error checking and minor refactoring

### PR DESCRIPTION
**Pull request by RyanG10**

### CODE REVIEW

File: src/components/visualizer/visualizerController.ts [hunk 1]
```diff
@@ -8,6 +8,24 @@ interface VisualizerControllerOptions {
...
+  VC_ERROR_DOUBLELOCK: -12,
+  VC_ERROR_CROSSTHREADUNLOCK: -13,
+  VC_ERROR_SEMVALUELIMIT: -14,
+  VC_ERROR_DEADLOCK: 15,
+};
+
+function assertUnreachable(e: never): never {
+  throw new Error('Unhandled case detected (assertUnreachable)');
+}
+function getErrorMsg(state: VC_ErrorCode): string {
```
Comment (TheIceCreamBear) [inline src/components/visualizer/visualizerController.ts:21]: Nit: add a new line above this function dec

File: src/components/visualizer/visualizerController.ts [hunk 2]
```diff
@@ -58,10 +76,17 @@ export default class VisualizerController {
   }
 
   private async startAsync() {
-    this.debuggerHandle = await window.platform.launchProgram(this.exeFile, data => {
+    const handle = await window.platform.launchProgram(this.exeFile, data => {
       //console.log("Console stdout output", data);
       this.onConsoleOutput([data]);
     });
+    if ('err' in handle) {
```
Comment (TheIceCreamBear) [inline src/components/visualizer/visualizerController.ts:83]: Not super familiar with this syntax, but is this essentially checking if the property `err` is present in `handle`?

**Review by TheIceCreamBear — COMMENTED** [review]
Looks good so far, the PR is still a draft so this isnt an explicit approval but it looks good

File: src/components/visualizer/visualizerController.ts (hunk 2 above)
Comment (RyanG10) [reply src/components/visualizer/visualizerController.ts:85]: Yes, that's correct

**Review by TheIceCreamBear — APPROVED** [review]
Looks good. Can I merge this?

**Comment by RyanG10** [comment]
Yes, I think so. But to be honest, it hasn't been tested very thoroughly.

**Comment by TheIceCreamBear** [comment]
Sounds good, ill merge it for future version v0.12.0

**Pull request merged**